packages/react-components/src/shared/src/useFocusScope.ts

```diff
@@ -16,27 +17,27 @@ class DomScope {
         return this._scopeRef.current;
     }
 
-    registerChangeHandler(handler) {
+    registerChangeHandler(handler: ((elements: T[], scope: T[]) => void)) {
         this._handlersRef.current.push(handler);
     }
 
-    removeChangeHandler(handler) {
+    removeChangeHandler(handler: ((elements: T[], scope: T[]) => void)) {
         const handlers = this._handlersRef.current;
 
         handlers.splice(handlers.indexOf(handler), 1);
     }
 
-    isInScope(element) {
+    isInScope(element: T) {
         return this.elements.some(x => x.contains(element));
     }
 }
 
 export function useFocusScope() {
-    const [scopeRef, setScope] = useRefState([]);
-    const [handlersRef] = useRefState([]);
+    const [scopeRef, setScope] = useRefState<HTMLElement[]>([]);
+    const [handlersRef] = useRefState<((elements: Element[], scope: Element[]) => void)[]>([]);
```

Comment:
Should extract a type for this since it's quite complicated and re-used in DomScope